src/main/java/io/pinecone/configs/PineconeConfig.java

```diff
@@ -50,17 +56,37 @@ public interface CustomChannelBuilder {
         ManagedChannel buildChannel();
     }
 
-    void validate() {
-        if (apiKey == null)
-            throw new PineconeValidationException("Invalid PineconeConfig: missing apiKey");
+    public void validate() {
+        if (apiKey == null || apiKey.isEmpty())
+            throw new PineconeConfigurationException("The API key is required and must not be empty or null");
     }
 
     public String getUserAgent() {
-        String userAgentLanguage = "lang=java; pineconeClientVersion = v0.8.0";
-        if (this.getIntegrationId() == null) {
+        String userAgentLanguage = "lang=java; pineconeClientVersion=v0.8.0";
+        if (this.getSourceTag() == null) {
             return userAgentLanguage;
         } else {
-            return userAgentLanguage + "; usageContext=" + this.getIntegrationId();
+            return userAgentLanguage + "; source_tag=" + this.getSourceTag();
+        }
+    }
+
+    private String normalizeSourceTag(String input) {
```

Comment:
We should add some unit tests around this normalization

Reply:
Yeah absolutely, not sure why I didn't. Will add some in, thanks!

Reply:
Unit tests added in `PineconeConfigTest.java`.